shenyu-client/shenyu-client-core/src/main/java/org/apache/shenyu/client/core/client/AbstractContextRefreshedEventListener.java

```diff
@@ -98,14 +98,14 @@ public AbstractContextRefreshedEventListener(final PropertiesConfig clientConfig
     
     @Override
     public void onApplicationEvent(@NonNull final ContextRefreshedEvent event) {
-        if (!registered.compareAndSet(false, true)) {
-            return;
-        }
         final ApplicationContext context = event.getApplicationContext();
         Map<String, T> beans = getBeans(context);
         if (MapUtils.isEmpty(beans)) {
             return;
         }
+        if (!registered.compareAndSet(false, true)) {
```

Comment:
why modify this？ 

when onApplicationEvent， getBeans(context) is null？

Reply:
yes